fighthealthinsurance/views.py

```diff
@@ -451,64 +447,80 @@ def ocr_upload(x):
+        }
+
+    def __init__(self, *args, **kwargs):
         self.regex_denial_processor = ProcessDenialRegex()
         self.codes_denial_processor = ProcessDenialCodes()
         self.regex_src = DataSource.objects.get(name="regex")
         self.codes_src = DataSource.objects.get(name="codes")
         self.zip_engine = uszipcode.search.SearchEngine()
 
-    def post(self, request):
-        form = DenialForm(request.POST)
-        if form.is_valid():
-            # It's not a password per-se but we want password like hashing.
-            # but we don't support changing the values.
-            email = form.cleaned_data["email"]
-            hashed_email = hashlib.sha512(email.encode("utf-8")).hexdigest()
-            denial_text = form.cleaned_data["denial_text"]
-            denial = Denial(denial_text=denial_text, hashed_email=hashed_email)
-            denial.save()
-            denial_types = self.regex_denial_processor.get_denialtype(denial_text)
-            denial_type = []
-            for dt in denial_types:
-                DenialTypesRelation(
-                    denial=denial, denial_type=dt, src=self.regex_src
-                ).save()
-                denial_type.append(dt)
-
-            plan_type = self.codes_denial_processor.get_plan_type(denial_text)
-            state = None
-            zip = form.cleaned_data["zip"]
-            if zip is not None and zip != "":
-                state = self.zip_engine.by_zipcode(form.cleaned_data["zip"]).state
-            (procedure, diagnosis) = appealGenerator.get_procedure_and_diagnosis(
-                denial_text
-            )
-            form = PostInferedForm(
-                initial={
-                    "denial_type": denial_type,
-                    "denial_id": denial.denial_id,
-                    "email": email,
-                    "your_state": state,
-                    "procedure": procedure,
-                    "diagnosis": diagnosis,
-                }
-            )
-            return render(
-                request,
-                "categorize.html",
-                context={
-                    "post_infered_form": form,
-                    "upload_more": True,
-                },
-            )
-        else:
-            return render(
-                request,
-                "scrub.html",
-                context={
-                    "error": form.errors,
-                    "ocr_result": request.POST.get("denial_text", ""),
-                },
-            )
+    def form_valid(self, form):
+        # It's not a password per-se but we want password like hashing.
+        # but we don't support changing the values.
+        email = form.cleaned_data["email"]
+        hashed_email = Denial.get_hashed_email(email)
+        denial_text = form.cleaned_data["denial_text"]
+
+        denial = Denial.objects.create(
+            denial_text=denial_text,
+            hashed_email=hashed_email,
+        )
+        denial.save()
```

Comment:
I think this was added, but we don't need to save here unless the **Denial** Model has some magic in its save method? The `create()` makes a database record itself.

Reply:
Oh yeah good point, I put it in while I was trying to figure out why it was not working but it was the fixtures.